v2/cmd/asoctl/internal/importing/resource_importer.go

```diff
@@ -74,18 +77,14 @@ func (ri *ResourceImporter) AddARMID(armID string) error {
 // This allows us to call it from other methods that already have the lock
 func (ri *ResourceImporter) addImpl(importer ImportableResource) {
 	// If we've already handled this resource, skip it
```

Comment:
Why didn't the previous implementation work?

If something is in the `pending` set or the `completed` set, we're already processing it (or processed it) and it doesn't need to be acted on again. Otherwise, it does?

Why do we also need `unique`?

Reply:
Because `completed` only lists resources we _successfully_ completed; If we have an error trying to handle the resource (say, `access denied`) then it didn't show up in `completed` and we retried (and refailed) every time we encountered it.

Reply:
Should `completed` be called `completedSuccessfully` or `successful` then? Obviously minor

Reply:
Renamed the field to `imported` (past tense).